### PR TITLE
chore: add form modifiers

### PR DESCRIPTION
**Pull request by shpran**
https://oat-sa.atlassian.net/browse/HKD-6

# Goal 

- Integration branch

# TODO 

- [ ] Update composer dependencies

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -64,10 +64,10 @@
     "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
     "qtism/qtism": ">=0.28.3",
     "oat-sa/generis": ">=15.36.4",
-    "oat-sa/tao-core": ">=54.21.0",
-    "oat-sa/extension-tao-item" : ">=12.1.0",
+    "oat-sa/tao-core": "dev-feat/ADF-1781/translations-feature as 99.99",
+    "oat-sa/extension-tao-item" : "dev-feat/ADF-1779/add-translator-role as 99.99",
     "oat-sa/extension-tao-itemqti" : ">=30.12.0",
-    "oat-sa/extension-tao-test" : ">=16.0.0",
+    "oat-sa/extension-tao-test" : "dev-feat/ADF-1779/add-translator-role as 99.99",
```
Comment (shpran) [inline composer.json:70]: - [ ] Update composer dependencies

**Review by gabrielfs7 — APPROVED** [review]
Thanks @shpran 


- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/Translation/Service/QtiIdentifierRetriever.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\Translation\Service;
+
+use core_kernel_classes_Resource;
+use Psr\Log\LoggerInterface;
+use taoQtiTest_models_classes_QtiTestService;
+
+class QtiIdentifierRetriever
```
Comment (gabrielfs7) [inline models/classes/Translation/Service/QtiIdentifierRetriever.php:29]: - [ ] @shpran Can we also add tests for this class, please?

Thanks!

File: models/classes/Translation/Service/QtiIdentifierRetriever.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        try {
+            $jsonTest = $this->qtiTestService->getJsonTest($test);
+        } catch (Throwable $exception) {
+            $this->logger->error('An error occurred while retrieving test data: ' . $exception->getMessage());
+
+            return null;
+        }
+
+        return json_decode($jsonTest, true)['identifier'] ?? null;
```
Comment (gabrielfs7) [inline models/classes/Translation/Service/QtiIdentifierRetriever.php:50]: @shpran I think the code can be simplified. I am also not sure we should silence that exception. If the identifier is not created, there is something wrong.

```suggestion
        try {            
            return json_decode($this->qtiTestService->getJsonTest($test), true)['identifier'] ?? null;
        } catch (Throwable $exception) {
            $this->logger->error('An error occurred while retrieving test data: ' . $exception->getMessage());

            throw $e;
        }
```

File: models/classes/Translation/Service/QtiIdentifierRetriever.php (hunk 2 above)
Comment (shpran) [reply models/classes/Translation/Service/QtiIdentifierRetriever.php:30]: Done ✅

File: models/classes/Translation/Service/QtiIdentifierRetriever.php (hunk 3 above)
Comment (shpran) [reply models/classes/Translation/Service/QtiIdentifierRetriever.php]: Done ✅

**Pull request merged**